### PR TITLE
add hover mouse event

### PR DESCRIPTION
**Pull request by wuboy0307**
I need a tooltip when mouseover a tree node.

### CODE REVIEW

File: example/cli/src/app/actions/actions.component.ts [hunk 1]
```diff
@@ -16,6 +16,14 @@ const actionMapping:IActionMapping = {
       $event.shiftKey
         ? TREE_ACTIONS.TOGGLE_ACTIVE_MULTI(tree, node, $event)
         : TREE_ACTIONS.TOGGLE_ACTIVE(tree, node, $event);
+    },
+    mouseover: (tree, node, $event) => {
```
Comment (adamkleingit) [inline example/cli/src/app/actions/actions.component.ts]: Can you change to mouseOver and mouseOut to be consistent with the other names?

File: lib/components/tree-node-wrapper.component.ts [hunk 2]
```diff
@@ -1,38 +1,40 @@
-import { Component, Input, ViewEncapsulation, TemplateRef } from '@angular/core';
+import { Component , Input , ViewEncapsulation , TemplateRef } from '@angular/core';
```
Comment (adamkleingit) [inline lib/components/tree-node-wrapper.component.ts]: Can you discard the extra whitespace and formatting?

File: lib/components/tree-node-wrapper.component.ts [hunk 3]
```diff
@@ -1,38 +1,40 @@
...
 import { TreeNode } from '../models/tree-node.model';
 
 @Component({
-  selector: 'tree-node-wrapper',
-  encapsulation: ViewEncapsulation.None,
-  styles: [],
+  selector: 'tree-node-wrapper' ,
```
Comment (adamkleingit) [inline lib/components/tree-node-wrapper.component.ts]: Same here

File: lib/components/tree-node-wrapper.component.ts [hunk 4]
```diff
@@ -1,38 +1,40 @@
...
+               (treeDropDragEnter)="node.mouseAction('dragEnter', $event)"
+               [treeAllowDrop]="node.allowDrop"
+               [allowDragoverStyling]="node.allowDragoverStyling()"
+               [treeDrag]="node"
+               [treeDragEnabled]="node.allowDrag()">
 
-          <tree-node-content [node]="node" [index]="index" [template]="templates.treeNodeTemplate">
-          </tree-node-content>
-        </div>
+              <tree-node-content [node]="node" [index]="index" [template]="templates.treeNodeTemplate">
```
Comment (adamkleingit) [inline lib/components/tree-node-wrapper.component.ts]: Same here

File: lib/components/tree-node-wrapper.component.ts [hunk 5]
```diff
@@ -1,38 +1,40 @@
...
+              </tree-node-content>
+          </div>
       </div>
-      <ng-container 
-        [ngTemplateOutlet]="templates.treeNodeWrapperTemplate" 
-        [ngTemplateOutletContext]="{ $implicit: node, node: node, index: index, templates: templates }">
+      <ng-container
+              [ngTemplateOutlet]="templates.treeNodeWrapperTemplate"
```
Comment (adamkleingit) [inline lib/components/tree-node-wrapper.component.ts]: Same here

File: example/cli/src/app/actions/actions.component.ts (hunk 1 above)
Comment (wuboy0307) [reply example/cli/src/app/actions/actions.component.ts]: "mouseover" and "mouseout" are angular's default event binding. The spelling is just like that (no camelcase). I prefer not to change that default naming to keep consistency.
However, I saw you also camelcase "dragOver". If that's the rule in this repository. I will still follow it.

Btw, there are also mouseenter and mouseleave (which events are not bubbling to children elements)

https://gist.github.com/AlexAegis/8741eb49db8112e6d0c05328d64a5e24

File: package.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "angular-tree-component",
```
Comment (adamkleingit) [inline package.json]: Why did you change the package name?

File: example/cli/src/app/actions/actions.component.ts (hunk 1 above)
Comment (adamkleingit) [reply example/cli/src/app/actions/actions.component.ts]: Yeah, I prefer to be consistent with the other actions. As these are not DOM events, they don't need to be consistent with Angular's event binding. In retrospect, it might have been better to keep it lowercase.
Thanks

**Comment by adamkleingit** [comment]
I updated to mouseOver and mouseOut and merged.
Thanks!
